backend/src/interface/image_hoster/model/mod.rs

```diff
@@ -0,0 +1,48 @@
+//! These structs are used for image hoster data
+use std::error::Error;
+use thiserror::Error;
+
+/// Struct contains information all necessary information that the image hoster provides.
+pub struct ImageMetaData {
+    /// The image hoster image identification.
+    pub id: String,
+    /// The url which short links to the image. Different to the url provided by an user.
+    pub image_url: String,
+    /// The licence of the image.
+    pub licence: String,
+}
+
+/// Enum describing the possible ways, a image hoster request can fail.
+#[derive(Debug, Error)]
+pub enum ImageHosterError {
```

Comment:
Do we really need all these detailed error messages?

Reply:
No, but I already created them. Some of them could be merged, to minimize this enum.